### DISCUSSION TITLE
Shells: support for quadratic triangular elements (6 nodes)

### DISCUSSION THREAD

**Discussion by jmeier** [idaholab/moose#23874] [Q&A Modules: Solid mechanics] [answered]
Dear Moose Comunity,

This question is abaout support for quadratic triangular elements (6 nodes) for shells.
Is it correct that Moose does not (yet) have tri6 shell elements ready? If not, are there plans to implement this?

In detail:

I have an MSH-File (gmsh) I import successfully in Moose. The msh contains volume-elements (TRI10), surface-elements (TRI6) and beam-Elements (EDGE3). Moose reads the elements and does not complain. So far, so good.
 
I tried to assign a shell-material:

```
[Materials]
  [./sign_elasticity]
    type = ADComputeIsotropicElasticityTensorShell
    youngs_modulus = 210E6
    poissons_ratio = 0.2
    block = ${volSign}
    through_thickness_order = SECOND
  []
  [./sign_strain]
    type = ADComputeIncrementalShellStrain
    block = ${volSign}
    displacements = 'disp_x disp_y disp_z'
    rotations = 'rot_x rot_y'
    thickness = 0.1
    through_thickness_order = SECOND
  []
  [./sign_stress]
    type = ADComputeShellStress
    block = ${volSign}
    through_thickness_order = SECOND
  []
[]
```

Now I get the following error:
```
*** ERROR ***
The following error occurred in the object "sign_strain", of type "ADComputeIncrementalShellStrain".
ADComputeIncrementalShellStrain: Shell element is implemented only for linear elements.
```

Thanks

Tag @cticenhour

**Comment by cticenhour** [comment]
Thanks for moving your question here, @grunerjmeier! Looking at the git blame, I am going to tag @sveerara for comment on this one. 

For general Tensor Mechanics knowledge, I'll also tag @bwspenc @sapitts.

**Reply by GiudGiud** [reply]
@recuero on shell elements

**Reply by GiudGiud** [reply]
@jain651

**Reply by recuero** [reply, marked as answer]
To my knowledge, there are no plans for extending the structural element support. The only available shell element is the one documented online: https://mooseframework.inl.gov/modules/tensor_mechanics/ShellElements.html.

**Reply by jmeier** [reply]
Thanks for your answer @recuero.
